MaraBot/Core/CommandUtils.cs

```diff
@@ -324,6 +328,158 @@ await MentionRoleWithoutPing(ctx, guildRoles.Select(r => r.Value).ToArray())
+                return AttachmentFileType.JsonPreset;
+            }
+            // Log Txt File
+            else if (Path.GetExtension(attachment.FileName).ToLower() == ".txt")
+            {
+                var body = attachment.FileName.Remove(attachment.FileName.Length - 4);
+
+                // Should not be the spoiler log.
+                if (body.EndsWith("_SPOILER"))
+                {
+                    errorMessage = "Spoiler log file has been provided. Please, provide non-spoiler log instead.";
+                    return AttachmentFileType.Invalid;
+                }
+
+                var regex = new Regex("log_[A-F0-9]{16}");
+                var match = regex.Match(body);
+
+                // Should match the usual log file format.
+                if (match.Length != body.Length)
+                {
+                    errorMessage = "Spoiler log file name is not standard. Expecting a log_ prefix followed by a 16 digits hexadecimal seed.";
+                    return AttachmentFileType.Invalid;
+                }
+
+                errorMessage = String.Empty;
+                return AttachmentFileType.LogFile;
+            }
+
+            errorMessage = "Unrecognized file extension";
+            return AttachmentFileType.Invalid;
+        }
+
+        private static async Task<Preset> LoadPresetAttachmentAsync(CommandContext ctx, IReadOnlyDictionary<string, Option> options)
+        {
+            var attachment = ctx.Message.Attachments[0];
+            var url = attachment.Url;
+
+            var request = WebRequest.Create(url);
+            var response = await request.GetResponseAsync();
+            var dataStream = response.GetResponseStream();
+
+            using (StreamReader r = new StreamReader(dataStream))
+            {
+                var jsonContent = await r.ReadToEndAsync();
+                var preset = PresetIO.LoadPreset(jsonContent, options);
+                if (preset.Equals(default))
+                    throw new InvalidOperationException("Could not parse custom json preset. Please supply a valid json file.");
+
+                return preset;
+            }
+        }
+
+        private static async Task<(Preset Preset, string Seed)> LoadLogAttachmentAsync(CommandContext ctx, IReadOnlyDictionary<string, Option> options)
+        {
+            var attachment = ctx.Message.Attachments[0];
+            var url = attachment.Url;
+
+            var request = WebRequest.Create(url);
+            var response = await request.GetResponseAsync();
+            var dataStream = response.GetResponseStream();
+
+            using (StreamReader r = new StreamReader(dataStream))
+            {
+                await r.ReadLineAsync();
+                var seedLineTask = r.ReadLineAsync();
+                var optionsLineTask = r.ReadLineAsync();
+                await r.ReadLineAsync();
+                await r.ReadLineAsync();
+                var versionLineTask = r.ReadLineAsync();
```

Comment:
I'm only reading the 6 first lines of the log files as everything we need is there. However, if Moppu decides to change format, this will not work anymore, so I'm not quite sold on this solution either.

We could read the whole file and run a regex on the entire file string instead, but the log files are quite big, so it's not ideal either.

The better way might be to read a "header" chunk of the log file and then run regex on this chunk alone.